src/Parsing/Impl/Ast/ScopeInfo.cs

```diff
@@ -0,0 +1,297 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Python.Core;
+using Microsoft.Python.Parsing.Definition;
+
+namespace Microsoft.Python.Parsing.Ast {
+    public abstract class ScopeInfo {
```

Comment:
Did you modify the binding logic at all to handle comprehensions, or is this just the same logic but relocated? As noted in #1586, I'm not sure the binder was able to handle comprehension scoping properly, but if you're fixing it, the bulk of #1586 might be able to be removed in a future PR.

Reply:
Nope all I handled was creating a new scope for comprehensions + framework for any Node. I could do that in a future PR